File: internal/domain/brands/brand.go
package brands

import "dresser/internal/domain/categoriess"

type ID int

type Brand struct {
	ID         ID
	Name       string
	Categories categoriess.Categories
}

func New(id int, name string, categories categoriess.Categories) (*Brand, error) {
	return &Brand{
		ID:         ID(id),
		Name:       name,
		Categories: categories,
	}, nil
}

func (b *Brand) AddCategory(category categoriess.Category) error {
	return b.Categories.Add(category)
}

func (b *Brand) RemoveCategory(category categoriess.Category) error {
	return b.Categories.Remove(category)
}

func (b *Brand) GetID() int {
	return int(b.ID)
}

func (b *Brand) GetName() string {
	return b.Name
}

func (b *Brand) GetCategories() []string {
	return b.Categories.Values()
}

File: java/src/main/java/com/dresser/domain/brands/Brand.java
package com.dresser.domain.brands;

public class Brand {
    private final BrandId id;
    private String name;
    
    private Brand(BrandId id, String name) {
        this.id = id;
        this.name = name;
    }
    
    public static Brand create(int id, String name) {
        return new Brand(new BrandId(id), name);
    }
    
    public void update(String name) {
        this.name = name;
    }
    
    // Getters
    public BrandId getId() {
        return id;
    }
    
    public String getName() {
        return name;
    }
    
    // Setters
    public void setName(String name) {
        this.name = name;
    }
    
    // For equality checking
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Brand brand = (Brand) o;
        return id.equals(brand.id);
    }
    
    @Override
    public int hashCode() {
        return id.hashCode();
    }
} 